File: languages/erlang/exercises/concept/basics/basics_test.erl
-module(lasagna_tests).
-include_lib("eunit/include/eunit.hrl").

expected_minutes_in_oven_test() ->
  ?assertEqual(40, lasagna:expected_minutes_in_oven()).

remaining_minutes_in_oven_test() ->
  ?assertEqual(15, lasagna:remaining_minutes_in_oven(25)).

preparation_time_in_minutes_test() ->
  ?assertEqual(2, lasagna:preparation_time_in_minutes(1)),
  ?assertEqual(8, lasagna:preparation_time_in_minutes(4)).

total_time_in_minutes_test() ->
  ?assertEqual(32, lasagna:total_time_in_minutes(1, 30)),
  ?assertEqual(16, lasagna:total_time_in_minutes(4, 8)).

File: languages/x86-64-assembly/exercises/concept/basics/basics_test.c
#include <assert.h>
#include <inttypes.h>

int64_t expected_minutes_in_oven(void);
int64_t remaining_minutes_in_oven(int64_t actual_minutes_in_oven);
int64_t preparation_time_in_minutes(int64_t number_of_layers);
int64_t total_time_in_minutes(int64_t number_of_layers, int64_t actual_minutes_in_oven);

int main(void) {
    assert(expected_minutes_in_oven() == 40);
    assert(remaining_minutes_in_oven(25) == 15);
    assert(preparation_time_in_minutes(1) == 2);
    assert(preparation_time_in_minutes(4) == 8);
    assert(total_time_in_minutes(1, 30) == 32);
    assert(total_time_in_minutes(4, 8) == 16);
}
